### PR TITLE
Signals cleanup

### PR DESCRIPTION
**Pull request by Rohesie**
* Moves signals to the base of the procs and adds a requirement to call the parent.
* Adds item attack signals and cleans up other existing ones.
* Cleans up scout cloak code.
* Removes procs that don't do anything.
* Deletes an unused define.

### CODE REVIEW

File: code/game/objects/items/storage/backpack.dm [hunk 1]
```diff
@@ -610,7 +609,19 @@
 		to_chat(wearer, "<span class='danger'>Your cloak shimmers from the damage!</span>")
 		apply_shimmer()
 
-/obj/item/storage/backpack/marine/satchel/scout_cloak/proc/action_taken(datum/source, atom/target, obj/item/I, mob/living/wearer)
+/obj/item/storage/backpack/marine/satchel/scout_cloak/proc/on_gun_firing(datum/source, atom/target, obj/item/weapon/gun/firing_gun)
+	action_taken()
+
+/obj/item/storage/backpack/marine/satchel/scout_cloak/proc/on_gun_attachment_firing(datum/source, atom/target, obj/item/attachable/firing_attachment, obj/item/weapon/gun/master_gun)
```
Comment (spookydonut) [inline code/game/objects/items/storage/backpack.dm]: these are pointless, if you dont care about the arguments just use one proc like it was before

File: code/modules/projectiles/updated_projectiles/gun_attachables.dm [hunk 2]
```diff
@@ -267,7 +267,8 @@ Defined in conflicts.dm of the #defines folder.
 	return
 
 /obj/item/attachable/proc/fire_attachment(atom/target,obj/item/weapon/gun/gun, mob/user) //For actually shooting those guns.
-	return
+	SHOULD_CALL_PARENT(1)
+	SEND_SIGNAL(user, COMSIG_HUMAN_ATTACHMENT_FIRED, target, src, gun)
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/gun_attachables.dm]:
```suggestion
	SEND_SIGNAL(user, COMSIG_MOB_ATTACHMENT_FIRED, target, src, gun)
```

File: code/modules/projectiles/updated_projectiles/gun_system.dm [hunk 3]
```diff
@@ -681,7 +681,7 @@ and you're good to go.
 			extra_delay = min(extra_delay+(burst_delay*2), fire_delay*3) // The more bullets you shoot, the more delay there is, but no more than thrice the regular delay.
 			sleep(burst_delay)
 
-		SEND_SIGNAL(user, COMSIG_HUMAN_GUN_FIRED, target, src, user)
+		SEND_SIGNAL(user, COMSIG_HUMAN_GUN_FIRED, target, src)
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/gun_system.dm]:
```suggestion
		SEND_SIGNAL(user, COMSIG_MOB_GUN_FIRED, target, src)
```

File: code/game/objects/items/storage/backpack.dm (hunk 1 above)
Comment (Rohesie) [reply code/game/objects/items/storage/backpack.dm]: Clarity. The people seeing the proc need to know correctly what parameters they are actually receiving, even if they are not currently being used.

Comment (spookydonut) [reply code/game/objects/items/storage/backpack.dm]: I disagree. someone else can merge this as I won't.

Comment (ninjanomnom) [reply code/game/objects/items/storage/backpack.dm]: It's up to the individual if they want to make a different proc for each signal if the alternative is a receiving proc that takes 0 (or 1 for the source) arguments

File: code/game/objects/items/storage/backpack.dm [hunk 4]
```diff
@@ -500,17 +500,14 @@
 
 	addtimer(CALLBACK(src, .proc/on_cloak), 1)
 	RegisterSignal(M, COMSIG_HUMAN_DAMAGE_TAKEN, .proc/damage_taken)
-	RegisterSignal(M, list(
-		COMSIG_HUMAN_GUN_FIRED,
-		COMSIG_HUMAN_ATTACHMENT_FIRED,
-		COMSIG_HUMAN_ITEM_THROW,
-		COMSIG_HUMAN_ITEM_ATTACK), .proc/action_taken)
+	RegisterSignal(M, list(COMSIG_MOB_GUN_FIRED, COMSIG_MOB_GUN_AUTOFIRED, COMSIG_MOB_ATTACHMENT_FIRED, COMSIG_MOB_THROW, COMSIG_MOB_ITEM_ATTACK), .proc/action_taken)
```
Comment (spookydonut) [inline code/game/objects/items/storage/backpack.dm]: formatting

**Pull request merged**